src/components/ProviderSettings/ProviderSettings.js

```diff
@@ -0,0 +1,61 @@
+// @flow
+import React from 'react';
+import ProviderSettingsRenderer from './ProviderSettingsRenderer';
+import type { ProviderOptions } from '../../types/provider';
+
+type Props = {
+  loading: boolean,
+  error: string,
+  currentProvider: ProviderOptions,
+  setProvider: ProviderOptions => void,
+};
+
+type State = ProviderOptions;
+
+class ProviderSettings extends React.PureComponent<Props, State> {
+  state = {
+    provider: '',
+    type: '',
+    networkId: 1,
+    url: '',
+  };
+
+  handleSubmit = (e: SyntheticEvent<>) => {
+    e.preventDefault();
+    this.props.setProvider(this.state);
+  };
+
+  handleChange = ({ target }: SyntheticInputEvent<>) => {
+    const value = target.type === 'checkbox' ? target.checked : target.value;
+    this.setState({ ...this.state, [target.name]: value });
+  };
+
+  handleNetworkChange = (network: Object) => {
+    this.setState({ ...this.state, networkId: network.id });
```

Comment:
You don't need to do `...this.state` within `this.setState()` as it will be merged together by the react internals.

Reply:
Ah very cool, i didn't know or didn't remember that ;p